cpu/avr8_common/include/delay_arch.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2023 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_avr8_common
+ * @{
+ *
+ * @file
+ * @brief       AVR specific busy delay
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@posteo.net>
+ *
+ */
+
+#ifndef DELAY_ARCH_H
+#define DELAY_ARCH_H
+
+#include <stdint.h>
+
+#include "board.h"
+#include "periph_conf.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+#ifndef DOXYGEN
+
+#define HAS_DELAY_US_SPINNING
+
+static inline void delay_us_spinning(uint32_t usecs)
+{
+    uint32_t cycles;
+    if (usecs <= (UINT32_MAX / CLOCK_CORECLOCK)) {
+        /* multication won't overflow 32 bit math */
+        cycles = usecs * (uint32_t)CLOCK_CORECLOCK / 1000UL;
+    }
+    else {
+        cycles = usecs * (uint64_t)CLOCK_CORECLOCK / 1000ULL;
+    }
```

Comment:
according to https://gcc.gnu.org/onlinedocs/gcc/AVR-Built-in-Functions.html __builtin_avr_delay_cycles(cycles) works only for compile-time-constant values => everything that leads to cycles must also be compile-time-constant  => no need for two different  multiplication and division implementations